Feature/first screen login

tm210608 opened the pull request:
* He creado todos los componentes de la pantalla del login, incluidos en el LoginScreen.kt.
* He aplicado los estados de los componentes necesarios, a través de LiveData, en LoginViewModel.kt
* He cargado las librerías necesarias en el gradle del módulo Login.
* Todos éstos cambios éstos cambios están realizados dentro del módulo Login del proyecto.

* He lanzado el LoginScreen en el Activity del módulo App.

tm210608 committed:
Preparación primera pantalla de login

tm210608 committed:
finalización de la pantalla de login

Cambios en el viewModel y activity del modulo login

tm210608 committed:
Update LoginScreen.kt

tm210608 committed:
Actualizacion first_screen_login

tm210608 committed:
Remove .idea

tm210608 committed:
Merge branch 'refs/heads/develop' into feature/first_screen_login

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/java/com/mito/fullapplication/MainActivity.kt
#	build.gradle.kts
#	gradle/libs.versions.toml
#	settings.gradle.kts

magijon commented on login/src/main/AndroidManifest.xml:
```diff
@@ -0,0 +1,13 @@
+
+    <application>
+        <activity
+            android:name=".ui.MainActivityLogin"
+            android:exported="false" />
+        <activity
+            android:name=".MainLoginActivity"
+            android:exported="false" />
```
Por que hay dos activities?

magijon commented on login/src/main/java/com/mito/login/ui/LoginScreen.kt:
```diff
@@ -0,0 +1,197 @@
+fun LoginScreenPreview(){
+    LoginScreen(viewModel = LoginViewModel())
+}
+//@Preview(showBackground = true, showSystemUi = true)
+//@Composable
+//fun EmailItemPreview(){
+//    EmailItem()
+//}
```
Hay que eliminar las lineas comentadas

magijon commented on login/src/main/java/com/mito/login/ui/LoginScreen.kt:
```diff
@@ -0,0 +1,197 @@
+    val password : String by viewModel.password.observeAsState(initial = "")
+    val loginEnable : Boolean by viewModel.loginEnable.observeAsState(initial = false)
+    val isLoading: Boolean by viewModel.isLoading.observeAsState(initial = false)
+    val coroutineScope = rememberCoroutineScope()
```
Esto se queda de momento... Pero al tener el ViewModel vamos a hacer que las corrutinas se lancen desde alli, no desde la capa de vista. La idea es que la lógica vaya en el ViewModel. Se que el video lo explica asi, pero es para enseñar un acercamiento.

magijon commented on login/src/main/java/com/mito/login/ui/LoginScreen.kt:
```diff
@@ -0,0 +1,197 @@
+            .fillMaxSize()
+            .padding(16.dp)
+    ){
+      Login(Modifier.align(Alignment.Center), viewModel)
```
Lo dejamos de momento, pero vamos a resposabilizar un poco más las cosas y para lo siguiente que corrijamos, no vamos a pasarle el ViewModel, vamos a pasarle actions y status.

magijon commented on login/src/main/java/com/mito/login/ui/LoginScreen.kt:
```diff
@@ -0,0 +1,197 @@
+        value = email,
+        onValueChange = {onTextFieldChanged(it)},
+        modifier = Modifier.fillMaxWidth(),
+        placeholder = { Text( text = "Email: Escriba aquí") },
```
Recurso de string

magijon commented on login/src/main/java/com/mito/login/ui/LoginScreen.kt:
```diff
@@ -0,0 +1,197 @@
+fun MainImage(modifier: Modifier) {
+    Image(
+        painter = painterResource(id = R.drawable.main_image),
+        contentDescription = "imagen de cabecera",
```
Otro recurso de string

magijon commented on login/src/main/java/com/mito/login/ui/LoginScreen.kt:
```diff
@@ -0,0 +1,197 @@
+    TextField(
+        value = password,
+        onValueChange = {onTextFieldChanged(it)},
+        placeholder = { Text(text = "Contraseña: Escriba aquí") },
```
Recurso de string

magijon commented on login/src/main/java/com/mito/login/ui/LoginScreen.kt:
```diff
@@ -0,0 +1,197 @@
+@Composable
+fun ForgotPassword(modifier: Modifier) {
+    Text(
+        text = "¿Olvidaste la contraseña?",
```
Recurso de string

magijon commented on login/src/main/java/com/mito/login/ui/LoginScreen.kt:
```diff
@@ -0,0 +1,197 @@
+        enabled = loginEnable,
+        shape = RoundedCornerShape(8.dp)
+    ){
+        Text(text = "Iniciar Sesión")
```
Los strings que metamos iran en el archivo de resources string.xml. Puedes modificar este y usar el recurso.

magijon commented on login/src/main/res/drawable/img.png:
Este archivo si no se va a usar debe borrarse

magijon commented on login/src/main/res/values/strings.xml:
```diff
@@ -0,0 +1 @@
+<resources></resources>
```
Aqui hay que meter todos los string resources. Intenta ponerle los nombres en minusculas todos, separados con guiones bajos, describiendo la pantalla donde va, el componente que rellena y una descripcion.

tm210608 committed:
feature/first_screen_login

-Quitado preview, img.png y LoginActivity
-Creados recursos de strings y cambiados en los textFields correspondientes del LoginScreen
-Cambios en los manifest de los módulos Login y App
